ManageIQ/manageiq
# Move authentication configuration to authselect in place of authconfig

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1707537

In RHEL8/CentOS8 authconfig is being replaced by authselect starting.

This PR updates the miqldap_to_sssd config tool to use authselect when available an to revert back to authconfig otherwise. This approach enables the miqldap_to_sssd tool to be agnostic to the OS upgrade to RHEL8/CentOS8 and will do the correct thing regaring authconfig/authselect on RHEL8/CentOS8 and earlier.

The approach was to:
  
rename the AuthConfig class to AuthEstablish
adding logic to AuthEstablish to invoke authselect when available and authconfig otherwise.
Add class AuthTemplateFiles to be shared by ConfigureApache and SssdConf as both now deal with configuration files and .erbs
Add SssdConf#create method to create the sssd.conf file as seeded by the sssd.conf.erb made available in a separate PR (see below)

Links [Optional]
----------------

[PR to add TEMPLATE/x/sssd.conf.erb](https://github.com/ManageIQ/manageiq-appliance/pull/234)


Steps for Testing/QA [Optional]
-------------------------------

Configure two appliances. One built on RHEL8 and the other on RHEL7
Configure each for Authentication mode LDAP/Get Groups From LDAP.
Run the miqldap_to_sssd conversion tool
Run the following command on the RHEL8 based test VM
> grep authselect /var/www/miq/vmdb/log/miqldap_to_sssd.log 

The output should look similar to:
> DEBUG: Ran command: authselect select sssd --force

Run the following command on the RHEL7 based test VM
> grep authconfig /var/www/miq/vmdb/log/miqldap_to_sssd.log 

The output should look similar to:
> DEBUG: Ran command: authconfig <params... bla bla bla>
